hazelcast/src/main/java/com/hazelcast/jet/Job.java

```diff
@@ -349,4 +350,20 @@ default String getName() {
      * @throws UnsupportedOperationException if called for a light job
      */
     JobStateSnapshot exportSnapshot(String name);
+
+    /**
+     * Associates the given listener to this job.
+     * @return The registration id
+     *
+     * @since 5.3
+     */
+    UUID addStatusListener(@Nonnull JobListener listener);
```

Comment:
You can add that as a code comment to the code (not javadoc).

Reply:
`Registration#equals` used to compare all fields including the ID, but it was modified to compare ID only in [#11520][1]. Including ID in the equality comparison makes it too restrictive as it causes the same _(service, topic, filter, listener)_ tuple to be seen as distinct for each `registerListener` call. However, the modification was done to avoid duplicate registrations during topological changes. In that case, the equality comparison of _filter_ and _listener_ cannot be done correctly, I guess, due to (de)serialization and ID is the only reliable field to reject the same tuple. Can someone from the @hazelcast/core team clarify this behavior? 

[1]: https://github.com/hazelcast/hazelcast/pull/11520/files#diff-a35624c3a555a7b2b4900f40b3b9d3513cad4d13a7caca3362fbccdb8d7ede03

Reply:
Ok, but add this comment as a source comment, if anyone wonders why there's no `@Nonnull` annotation.

Reply:
It's added to the `EventServiceImpl#registerListener0`, which is the source of the problem.